pandas/_libs/index.pyx

```diff
@@ -491,6 +494,76 @@ cdef class TimedeltaEngine(DatetimeEngine):
     cdef _get_box_dtype(self):
         return 'm8[ns]'
 
+
+cdef class PeriodEngine(Int64Engine):
+
+    cdef _get_index_values(self):
+        return self.vgetter()
+
+    cpdef _call_map_locations(self, values):
```

Comment:
is this necessary to create another way of calling this? e.g. ``_call_map_locations``

Reply:
@jreback So you mean that we don't have to define `_call_map_locations` by `cpdef`?

Reply:
@jreback I defined `_call_map_locations` as `cdef`, but some tests fail.
Seems that `_ensure_mapping_populated` is `inline`.